src/pool/srv_pool.c

```diff
@@ -4621,19 +4633,23 @@ ds_pool_evict_handler(crt_rpc_t *rpc)
 }
 
 /**
- * Send a CaRT message to the pool svc during pool destroy to test and
- * (if applicable based on force option) evict all open handles on a pool.
+ * Send a CaRT message to the pool svc to test and
+ * (if applicable based on destoy and force option) evict all open handles
+ * on a pool.
  *
  * \param[in]	pool_uuid	UUID of the pool
  * \param[in]	ranks		Pool service replicas
+ * \param[in]	destroy		If true the evict request is a destroy request
```

Comment:
Can you explain a bit more about the destroy parameter? Does this destroy the pool after evicting the handles?

Reply:
Just some input to the discussion, will let @dpquigl answer with respect to this change. Original addition of this function and its role in pool destroy flow is noted in commit message of [070209a4fa4f5f516238d6a51cbee0056a86b1d1](https://github.com/daos-stack/daos/commit/070209a4fa4f5f516238d6a51cbee0056a86b1d1)

Reply:
In the original code there is a conditional that checks if the request being processed came from a pool destroy. It was always hard coded to 1 for some odd reason. We needed a way to trigger the evict code without going down the destroy path.

Reply:
It was hard coded to 1 because the ds_pool_svc_check_evict() was written originally to be run only in the pool destroy execution flow, to perform pool handle evictions as applicable. The function naming did not make that very obvious at the time. However it may be fortunate here in that it can be generalized and used for both evict and destroy.

Reply:
What is odd is that it was still being used on the pool evict path. So I'm guessing the final destroy portion of the command was done in the destroy handler and it was only being used to evict the handles?

Reply:
Originally it was used in the destroy path yes just to evict handles with the rest of the destroy processing handled elsewhere. More recently with a patch implementing the MS in the control plane, this was added to the pool evict execution path.